daemon/graphdriver/windows/windows.go

```
@@ -0,0 +1,326 @@
+}
+
+// ApplyDiff extracts the changeset from the given diff into the
+// layer with the specified id and parent, returning the size of the
+// new layer in bytes.
+func (d *WindowsGraphDriver) ApplyDiff(id, parent string, diff archive.ArchiveReader) (size int64, err error) {
+	start := time.Now().UTC()
+	logrus.Debugf("WindowsGraphDriver ApplyDiff: Start untar layer")
+
+	destination := d.dir(id)
+	if d.info.Flavour == diffDriver {
+		destination = filepath.Dir(destination)
+	}
+
+	if size, err = chrootarchive.ApplyLayer(destination, diff); err != nil {
+		return
+	}
+	logrus.Debugf("WindowsGraphDriver ApplyDiff: Untar time: %vs", time.Now().UTC().Sub(start).Seconds())
+
+	return
+}
+
+// DiffSize calculates the changes between the specified layer
+// and its parent and returns the size in bytes of the changes
+// relative to its base filesystem directory.
+func (d *WindowsGraphDriver) DiffSize(id, parent string) (size int64, err error) {
+	changes, err := d.Changes(id, parent)
+	if err != nil {
+		return
+	}
+
+	layerFs, err := d.Get(id, "")
+	if err != nil {
+		return
+	}
+	defer d.Put(id)
+
+	return archive.ChangesSize(layerFs, changes), nil
+}
+
+func (d *WindowsGraphDriver) CopyDiff(sourceId, id string, parentLayerPaths []string) error {
+	d.Lock()
+	defer d.Unlock()
+
+	if d.info.Flavour == filterDriver && d.active[sourceId] == 0 {
+		if err := hcsshim.ActivateLayer(d.info, sourceId); err != nil {
+			return err
+		}
+		defer func() {
+			err := hcsshim.DeactivateLayer(d.info, sourceId)
+			if err != nil {
+				logrus.Warnf("Failed to Deactivate %s: %s", sourceId, err)
+			}
+		}()
+	}
+
+	return hcsshim.CopyLayer(d.info, sourceId, id, parentLayerPaths)
+}
+
+func (d *WindowsGraphDriver) LayerIdsToPaths(ids []string) []string {
+	var paths []string
+	for _, id := range ids {
+		path, err := d.Get(id, "")
+		if err != nil {
+			logrus.Debug("LayerIdsToPaths: Error getting mount path for id", id, ":", err.Error())
```

Why not return the error here?